### PR TITLE
Rkcg for open boundary flow pull request

### PR DESCRIPTION
**Pull request by FengWang3119**
1. Make the RKCG applicable for OpenBoundaryFlow by Correcting Bj when j is truncated.
2. Correct the bug that in the DensitySummationFreestream, the volume is not updated.
3. Improve the pressure outlet to make it compatible with RKCG.

### CODE REVIEW

File: src/shared/particle_dynamics/fluid_dynamics/fluid_integration.hpp
Comment (Bo-Zhang1995) [inline src/shared/particle_dynamics/fluid_dynamics/fluid_integration.hpp:1]: the line 87 why the correction for pressure_i using index_i and index_j?

Comment (Bo-Zhang1995) [reply src/shared/particle_dynamics/fluid_dynamics/fluid_integration.hpp:1]: correction(index_j, index_i) will return B_[index_j], what is the difference against correction(index_j)?

File: src/shared/particle_dynamics/particle_functors.h [hunk 1]
```diff
@@ -199,13 +204,41 @@ class LinearGradientCorrection : public KernelCorrection
...
+class LinearGradientCorrectionWithBulkScope : public KernelCorrection
+{
+  public:
+    LinearGradientCorrectionWithBulkScope(BaseParticles *particles)
+        : KernelCorrection(),
+          B_(*particles->getVariableDataByName<Matd>("LinearGradientCorrectionMatrix")),
+          within_scope_(particles){};
+
+    Matd operator()(size_t index_j, size_t index_i)
+    {
```
Comment (Bo-Zhang1995) [inline src/shared/particle_dynamics/particle_functors.h:230]: I got it. This is the answer to my concern. There is a selection between i and j.

Comment (Xiangyu-Hu) [reply src/shared/particle_dynamics/particle_functors.h:230]: This is not satisfactory. However, at present no other good way to incorporate the change.
Actually, this case raise a issue on how to handle boundary like region within a body.  it is not other body, not ghost and need to handle separately,

File: src/shared/particle_dynamics/fluid_dynamics/fluid_integration.hpp
Comment (FengWang3119) [reply src/shared/particle_dynamics/fluid_dynamics/fluid_integration.hpp:1]: For ordinary case where j is not truncated, the result is the same.

**Comment by Xiangyu-Hu** [comment]
@FengWang3119 why you close the pull request?

**Pull request closed without merging**